src/sentry/lang/native/symbolicator.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import absolute_import
+
+import six
+
+import time
+import logging
+
+from django.core.urlresolvers import reverse
+
+from requests.exceptions import RequestException
+
+from sentry import options
+from sentry.cache import default_cache
+from sentry.utils import metrics
+from sentry.net.http import Session
+from sentry.tasks.store import RetrySymbolication
+
+MAX_ATTEMPTS = 3
+REQUEST_CACHE_TIMEOUT = 3600
+SYMBOLICATOR_TIMEOUT = 5
+
+logger = logging.getLogger(__name__)
+
+
+def run_symbolicator(stacktraces, modules, project, arch, signal, request_id_cache_key):
+    self_url_prefix = options.get('system.url-prefix')
+
+    assert self_url_prefix
+    self_bucket_url = '%s%s' % (
+        self_url_prefix.rstrip('/'),
+        reverse('sentry-api-0-dsym-files', kwargs={
+            'organization_slug': project.organization.slug,
+            'project_slug': project.slug
+        })
+    )
+
+    project_id = six.text_type(project.id)
+    request_id = default_cache.get(request_id_cache_key)
+    sess = Session()
+
+    attempts = 0
+    wait = 0.5
+
+    with sess:
+        while 1:
+            try:
+                rv = _do_send_request(
+                    sess=sess,
+                    request_id=request_id,
+                    project_id=project_id,
+                    self_bucket_url=self_bucket_url,
+                    signal=signal,
+                    stacktraces=stacktraces,
+                    modules=modules
+                )
+
+                metrics.incr('events.symbolicator.status.%s' % rv.status_code, tags={
+                    'project_id': project_id
+                })
+
+                if rv.status_code == 404 and request_id:
+                    default_cache.delete(request_id_cache_key)
+                    raise RetrySymbolication(retry_after=0)
```

Comment:
Can we "fall through" and call the full symbolication endpoint immediately instead?